src/Nethermind/Nethermind.Evm/Instructions/EvmInstructions.Call.cs

```diff
@@ -176,6 +177,14 @@ public static EvmExceptionType InstructionCall<TOpCall, TTracingInst>(
             !UpdateGas(gasExtra, ref gasAvailable))
             goto OutOfGas;
 
+        // Retrieve code information for the call and schedule background analysis if needed.
+        ICodeInfo codeInfo = vm.CodeInfoRepository.GetCachedCodeInfo(state, codeSource, spec);
+
+        // If contract is large, charge for access
+        if (spec.IsEip7907Enabled &&
+            !ChargeForLargeContractAccess(codeInfo, vm, ref gasAvailable))
```

Comment:
Maybe move `IsEip7907Enabled` check within the method?

```suggestion
        if (!ChargeForLargeContractAccess(codeInfo, vm, ref gasAvailable))
```